iceoryx_dds/include/iceoryx_dds/dds/cyclone_data_writer.hpp

```diff
@@ -26,27 +26,49 @@ namespace iox
 {
 namespace dds
 {
-///
 /// @brief Implementation of the DataWriter abstraction using the cyclonedds implementation.
-///
 class CycloneDataWriter : public iox::dds::DataWriter
 {
   public:
     CycloneDataWriter() = delete;
+
+    /// @brief Constructor to set cyclone data writer object from given IDs
+    /// @param[in] serviceId ID of the service
+    /// @param[in] instanceId ID of the instance of the service
+    /// @param[in] eventId ID of the event
     CycloneDataWriter(const capro::IdString_t serviceId,
                       const capro::IdString_t instanceId,
                       const capro::IdString_t eventId) noexcept;
+
     virtual ~CycloneDataWriter();
+
     CycloneDataWriter(const CycloneDataWriter&) = delete;
     CycloneDataWriter& operator=(const CycloneDataWriter&) = delete;
-    // Required for vector
-    CycloneDataWriter(CycloneDataWriter&&) = default;
-    CycloneDataWriter& operator=(CycloneDataWriter&&) = default;
 
+    /// @brief Default move constructor to set the cyclone data writer
```

Comment:
This is just weird. Why documenting compiler-generated methods? 
The compiler generates a lot for us see this example:

```cpp
struct Fuu {
   int a;
};
```

In this case the compiler generates  methods so that the struct `Fuu` looks like:
```cpp
struct Fuu {
   Fuu();
   Fuu(Fuu&&);
   Fuu(const Fuu&);
   Fuu& operator=(Fuu&&);
   Fuu& operator=(const Fuu&);
   ~Fuu();

   int a;
};
```

And when we implement `Fuu` we try to follow the rule of zero, and if not possible, the rule of five. But in the rule of zero case we would have 6 methods which wouldn't be documented - and it makes sense!

So please remove the documentation from all compiler generated methods. Those methods are documented by the C++ standard which describes how they behave, what they do and shouldn't do.